Add rollout information (#514)
JuliaDiff/ChainRulesCore.jl: docs/src/index.md

Note that the packages with rules do not have to depend on AD systems, and neither do the AD systems have to depend on individual packages.

## ChainRules roll-out status

> Approximately [70 packages](https://juliahub.com/ui/Packages/ChainRulesCore/G6ax7/1.11.1?page=2) depends on ChainRulesCore to define rules.

Reviewer: ```suggestion
Numerous [packages](https://juliahub.com/ui/Packages/ChainRulesCore/G6ax7/?page=2) depend on ChainRulesCore to define rules for their functions.
```
This URL always gets latest version AFAICT

> 6 AD engines use ChainRules to get access to rules:

Reviewer: ```suggestion
6 AD engines currently use ChainRules to get access to rules:
```

[Zygote.jl](https://github.com/FluxML/Zygote.jl) is a reverse-mode AD that supports using `rrule`s, calling back into AD, and opting out of rules.

> However, if defined, its own primitives (`@adjoint`s) take precedence before `rrule`s.

Reviewer: ```suggestion
However, its own [ZygoteRules.jl](https://github.com/FluxML/ZygoteRules.jl/) primitives (`@adjoint`s) take precedence before `rrule`s when both are defined -- even if the `@adjoint` is less specific than the `rrule`.
```

Internally it uses its own set of tangent types, e.g. `nothing` instead of `NoTangent`/`ZeroTangent`.
It also `unthunk`s every tangent.

[Diffractor.jl](https://github.com/JuliaDiff/Diffractor.jl) is a forward- and reverse-mode AD that fully supports ChainRules, including calling back into AD, opting out of rules, and uses tangent types internally.

> [Yota](https://github.com/dfdx/Yota.jl) is a reverse-mode AD that supports using `rrule`s, but not calling back into AD and opting out of rules.

Reviewer: Yota's support is actually pretty comprehensive.
```suggestion
[Yota](https://github.com/dfdx/Yota.jl) is a reverse-mode AD that fully supports ChainRules, including calling back into AD, opting out of rules, and uses tangent types internally.
```

Author: Oh, interesting! I took a quick look and couldn't find a `RuleConfig` defined anywhere so just assumed that it doesn't support that.

Reviewer: It's there

[ReverseDiff](https://github.com/JuliaDiff/ReverseDiff.jl) is a reverse-mode AD that supports using `rrule`s, but not calling back into AD and opting out of rules.

> [Nabla.jl](https://github.com/invenia/Nabla.jl) is a reverse-mode AD that supports using `rrule`s and opting out of rules, but not calling back into AD.

Reviewer: ```suggestion
[Nabla.jl](https://github.com/invenia/Nabla.jl) is a reverse-mode AD that supports using `rrule`s, but not opting out of rules, nor calling back into AD.
```

Author: We did discuss how it can be done in principle though, right? In that it could work simply by looking at the returned `nothing`?

Reviewer: yes, but it is hardish

> [ReversePropagation.jl](https://github.com/dpsanders/ReversePropagation.jl) is a reverse-mode AD that supports using `rrule`s, but not calling back into AD and opting out of rules.

Reviewer: ```suggestion
[ReversePropagation.jl](https://github.com/dpsanders/ReversePropagation.jl) is a reverse-mode AD that supports using `rrule`s for scalar functions, but not calling back into AD and opting out of rules.
```
